File: statistics/3_regression/6_ex.py
# -*- coding: utf-8 -*-

from unittest import result
import numpy as np
import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt
import math

plt.style.use('seaborn')
mpl.rcParams['font.family'] = 'serif'

import scipy.stats as stats

# (1)

# read the data
AvgTemp = pd.read_excel('data/AvgTemp.xls')

Year = AvgTemp['year'].values
AvgTmp = AvgTemp['AvgTmp'].values

# get the Pearson's correlation
stats.pearsonr(Year, AvgTmp)

# get the Spearman's rho
stats.spearmanr(Year, AvgTmp)

# get the Kendall's tau
stats.kendalltau(Year, AvgTmp)


# (2)
import statsmodels.formula.api as smf

resTMP = smf.ols('AvgTmp ~ year', data=AvgTemp).fit()

print(resTMP.summary())

# (3)
stats.kstest(resTMP.resid, 'norm')

File: statistics/3_regression/6_ex.r
library(tidyverse)

# (1)

# load the data
# > source:https://www.r-bloggers.com/2021/06/reading-data-from-excel-files-xlsxlsxcsv-into-r-quick-guide/
# use package readxl
install.packages("readxl")
library(readxl)
AvgTemp <- read_excel('data/AvgTemp.xls')

Year <- AvgTemp$year
AvgTmp <- AvgTemp$AvgTmp

# get the Pearson's correlation
cor.test(Year, AvgTmp, method="pearson")

# get the Spearman's rho
cor.test(Year, AvgTmp, method="spearman")

# get the Kendall's tau
cor.test(Year, AvgTmp, method="kendall")


# (2)
resTMP <- lm(AvgTmp ~ year, data=AvgTemp)
summary(resTMP)

# (3)
ks.test(resTMP$residuals, 'pnorm')
